### ISSUE TITLE
0.24.0-beta-3 incorrectly escapes unicode character sequences in iOS strings file

### ISSUE THREAD

**Issue by darronschall** [icerockdev/moko-resources#698]
I have this string defined in my multiplatform strings.xml file:

```
<string name="youre_doing_great">You’re doing great!</string>
```

Note that `’` in there; that's a unicode "Right Single Quotation Mark" character U+2019.

For iOS, this is generating in the Localizable.strings file as:

```
"youre_doing_great" = "You\u2019re doing great!";
```

This is showing up in my app as: "Youu2019re doing great!". The unicode escape sequence is not being properly escaped.

After some investigating, I found https://stackoverflow.com/questions/23452906/ios-localization-unicode-character-escape-sequences-which-have-the-form-uxxx with an accepted answer that says:

> Strings-files, on the other hand, use \Unnnn to denote a UTF-16 character, and "UTF-16 surrogate pairs" for characters > U+FFFF:

I think the fix here is to use `\U` when escaping unicode for iOS string file generation.

Android works fine. It generates as `<string name="youre_doing_great">You\u2019re doing great!</string>` and displays correctly in my testing.

(This was not an issue in 0.23.0, it's something 0.24.0 introduced).

**Comment by syxc** [comment]
I found the same bug in the new version.

String Resources:

`<string name="label_count">Count:</string>`

Get it in `iosMain` with the following code:

Note: We don't need compose multiplatform.

`val labelCount = MR.strings.label_count.desc().localized()`

In 0.24.0-beta-1, the result is: `Count:` ✅  
In 0.24.0-beta-2/3, the result is: `u8BA1u6570uFF1A` ❌

**Comment by diarmuidmcg** [comment]
Hello, I am having a similar issue, but relating to accent marks on certain characters.  
My french texts ("idées", "ingrédients", etc) are displaying as 
"idu00E9es" & "ingru00E9dients". 

I upgraded from `0.20.1 `to `0.24.0-beta-3`

Happy to provide more information if needed.

**Comment by Alex009** [comment]
fixed in 0.24.0-beta-4
